apollo-router/src/plugin/mod.rs

```diff
@@ -182,6 +182,10 @@ pub trait Plugin: Send + Sync + 'static {
     where
         Self: Sized;
 
+    /// Called once all plugins have been initialized and it is safe to assume that this plugin will take effect.
+    /// Interaction with globals is safe in this method.
+    fn activate(&mut self) {}
```

Comment:
Let us know how the "not adding activate() but having a crate private callback" idea works out.

Reply:
I'm probably not going to have time to tackle this until next week. Suggest that we merge as is and do a follow-up before the release lands that has this PR in.

Reply:
Seems sketchy... We'd have to remember that was a potential "release blocker" if we did that.

Reply:
We don't have a changeset at all for this yet, so will put this in.

Reply:
How about this: [fb7fdbf](https://github.com/apollographql/router/pull/2358/commits/fb7fdbfc95a3d0f3bc88c43e6d1bc55b843e14e4)

It removes activate in favour of downcasting. Slightly leaky, but way easier than the callback thing.

Reply:
I like the approach. I'm not sure it's API compatible if we move the existing downcast from Plugin to DynPlugin. Won't it work on Plugin?

Reply:
It was already on `DynPlugin`. I'm actually not sure why it was ever on `Plugin` in the first place. It was also only ever on `Plugin` in `cfg(test)`.

I removed it and everything still compiled.